chainstate/src/detail/error.rs

```diff
@@ -46,10 +46,12 @@ pub enum BlockError {
     CheckBlockFailed(#[from] CheckBlockError),
     #[error("Failed to update the internal blockchain state: {0}")]
     StateUpdateFailed(#[from] ConnectTransactionError),
+    // FIXME: remove this in favor of PropertyQueryError?
     #[error("Failed to load best block")]
     BestBlockLoadError(PropertyQueryError),
-    #[error("Failed to load block")]
-    BlockLoadError(PropertyQueryError),
+    #[error("Generic property query error")]
+    // FIXME: is #[from] a good idea, provided that we have several errors that can contain PropertyQueryError?
```

Comment:
If I understand correctly, you mean that we have different scenarios that return the same error, making their treatment different, where #[from] will make that seamless and cause ignoring it?

In that case, I'm OK with removeing that #[from], but please add a comment above this error explaining why we're not using #[from] to avoid falling into that mistake again.

Reply:
[discussed] Actually, it was vice versa - I added the PropertyQueryError enum variant here and was wondering whether having "#[from]" for it was a good idea. But having PropertyQueryError is itself not a good idea, so I reverted this change.